core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -68,7 +67,12 @@ public Suggest() {
     }
 
     public Suggest(List<Suggestion<? extends Entry<? extends Option>>> suggestions) {
+        // we sort suggestions by their names to ensure iteration over suggestions are consistent
+        // this is needed as we need to fill in suggestion docs in SearchPhaseController#sortDocs
+        // in the same order as we enrich the suggestions with fetch results in SearchPhaseController#merge
+        suggestions.sort((o1, o2) -> o1.getName().compareTo(o2.getName()));
```

Comment:
we need to sort because we use a `Map<String, List<Suggestion>>` to group suggestions when we merge suggestions in `SearchPhaseController#merge` and when we reduce suggestions in `SearchPhaseController#sortDocs`, so there is no guarantee that the order of suggestions in both cases would be consistent.